src/parser.rs

```diff
@@ -212,27 +212,36 @@ impl<'a> Parser<'a> {
 
     pub fn path(self) -> Result<'a, String> {
         let (path, parser) = self.word()?;
-        let mut path = Path::new(path);
 
-        // For Linux and BSD and MacOS.
+        let path = Path::new(path);
+        if path.to_str().is_none() {
+            return Ok((String::from(""), parser));
+        }
+
+        let mut path_string = String::from(path.to_str().unwrap());
+
+        // Linux and BSD and MacOS use ~ to infer the home directory of a given user
         if cfg!(unix) && path.starts_with("~") {
             if let Some(base_dirs) = dirs::BaseDirs::new() {
-                let home = base_dirs.home_dir().to_str().unwrap().to_owned();
 
                 if path == Path::new("~") {
-                    return Ok((home, parser));
+                    return Ok((base_dirs.home_dir().to_str().unwrap().to_string(),parser));
```

Comment:
Looking at this whole function again, it seems like we're using `to_str()` a lot. Why not keep everything as a `Path` until the very last moment, and then finally convert to a `String`?

Reply:
I've since rewritten everything in that function to utilize PathBuf et al.